### ISSUE TITLE
Is it possible to generate an HTML AST from Markdown?

### ISSUE THREAD

**Issue by njlr** [fsprojects/FSharp.Formatting#926]
I would like to parse some Markdown, convert it to HTML, embed that in some other HTML and render it all to an HTML string. 

Something like this:

```fsharp
// Not real code

let markdown = Markdown.Parse(content)
let markdownHtml = Markdown.toHtmlElement markdown

let page = 
  html
    []
    [
      div [] [ str "This is my header" ]
      markdownHtml
      div [] [ str "This is my footer" ]
    ]

printfn "%s" (Html.toString html)
```

From the docs, I can see that `FSharp.Formatting` supports the markdown parsing bit, and can convert to an HTML string, but does it allow one to work with the HTML AST?

The [HTML model](https://github.com/fsprojects/FSharp.Formatting/blob/main/src/FSharp.Formatting.Common/HtmlModel.fs#L301) appears to be `internal`.

**Comment by nhirschey** [comment]
No, not the HTML model. But you can use `FSharp.Formatting`'s internal representation, the [literate document](https://fsprojects.github.io/FSharp.Formatting/reference/fsharp-formatting-literate-literatedocument.html) type. Once you have the literate document you can manipulate it as you see fit (as you could with an HTML AST).

See https://fsprojects.github.io/FSharp.Formatting/markdown.html

**Comment by dsyme** [comment]
/repo-assist chime in here with an opinion and analysis please
